src/shared/components/igv/IntegrativeGenomicsViewer.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from "react";
+import igv from 'igv/dist/igv.esm.js';
+import autobind from "autobind-decorator";
+import {observable, reaction} from "mobx";
+
+type IGVProps = {
+    tracks?: any[]; // TODO add typedef for tracks?
+    width?: number;
+    locus?: string|string[];
+};
+
+export default class IntegrativeGenomicsViewer extends React.Component<IGVProps, {}> {
+
+    public static defaultProps = {
+        genome: "hg19",
+        locus: "all"
+    };
+
+    private igvDiv: HTMLDivElement|undefined;
+    private igvBrowser: any;
+
+    @observable private locus: string|string[]|undefined;
+
+    constructor(props: IGVProps) {
+        super(props);
+
+        reaction(
```

Comment:
this reaction isn't disposed so it will live (and react forever).  we should only use reactions as absolute last resort.  i think we can accomplish the same with react lifecycle hooks.   you could put this in componentDidUpdate.   or make componentSHouldUpdate= false and use componentWillRecieveProps to call igvBrowser.search (no locus observable prop necessary)